server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -90,9 +90,21 @@ public static class Defaults {
      */
     @PublicApi(since = "1.0.0")
     public enum Dynamic {
-        TRUE,
-        FALSE,
-        STRICT
+        TRUE("true"),
+        FALSE("false"),
+        STRICT("strict"),
+        STRICT_ALLOW_TEMPLATES("strict_allow_templates");
```

Comment:
I think this my last concern: so in mixed mode cluster (3.0 + 2.x, or 2.16.0 + 2.15.0), this change will (I assume) cause failures since new enum member won't be recognized by old clusters. So what we have to do is to change how the dynamic value is being serialized for these version: instead of sending `STRICT_ALLOW_TEMPLATES` the only sensible substitution we have is `STRICT`

Reply:
Index mapping is part of the cluster state, so it's managed by the cluster_manager node, in a mixed cluster, the cluster_manager must be the lower version node to avoid metadata incompatibility issue, so we cannot create an index with setting `dynamic` to `strict_allow_templates` in a mixed version cluster if the cluster_manager doesn't support that parameter, just like implement a new field type, we don't need to worry about that, I think.

Reply:
Got it, thank you!